### PR TITLE
CAMEL-16285: Extensible ResourceHelper

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus here a first sketch of a extensible resource loader
@gnodet I have changed the `SpiGeneratorMojo` to also handle nested classes, it does not seem to break anything ... so far

**Review by oscerd — APPROVED** [review]

File: components/camel-bean/src/main/java/org/apache/camel/language/bean/BeanResourceResolver.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package org.apache.camel.language.bean;
```
Comment (davsclaus) [inline components/camel-bean/src/main/java/org/apache/camel/language/bean/BeanResourceResolver.java:1]: License header missing

**Comment by davsclaus** [comment]
@lburgazzoli this looks really good

**Comment by lburgazzoli** [comment]
@davsclaus @oscerd @gnodet  the PR is now ready for review

**Review by davsclaus — APPROVED** [review]

**Pull request merged**